webapp/src/container.tsx

```diff
@@ -276,6 +277,12 @@ export class SettingsMenu extends data.Component<SettingsMenuProps, SettingsMenu
         this.props.parent.signOutGithub();
     }
 
+    onDropdownShow = () => {
+        // Ensure dropdown always has the editor's most up-to-date menu items
+        const extDownloadMenuItems = pxt.commands.getDownloadMenuItems?.() || [];
+        this.setState({ extDownloadMenuItems });
```

Comment:
instead of storing these in state, is it easier to just call the commands function again when rendering? caching like this seems like it might run into some bugs if things were to update...

Reply:
fixed